### DISCUSSION TITLE
Cron Job Solution

### DISCUSSION THREAD

**Discussion by Pandoks** [basecamp/kamal#1327] [Q&A]
Cron jobs have been a headache for me for quite some time now and I haven't figured out a simple solution. Note, I am not using Ruby on Rails but I'm deploying Node and Go servers with it. Currently I have accessories Postgres and Clickhouse with read replicas and shared and I also have a dedicated backup server Pgbackrest.

I want to schedule regular backups but I need to run `pgbackrest backup` command in the Pgbackrest container. I have other cron job needs but that's just an example.

How should I handle cron jobs? Install and use cron on each individual container that I need it on? That would complicate some of my setup as I need to horizontally scale some of the containers. Should I create a dedicated container that handles cron jobs? How would it access containers and containers across hosts?

**Comment by tuladhar** [comment]
Is "Pgbackrest" container running as a Kamal accessory? The recommended best practice is to run cron jobs on dedicated container, separate from the application container.
